code/messaging/src/main/java/com/adobe/marketing/mobile/MessagingPushUtils.java

```diff
@@ -0,0 +1,85 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile;
+
+import android.content.ContentResolver;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import androidx.annotation.NonNull;
+import com.adobe.marketing.mobile.services.Log;
+import com.adobe.marketing.mobile.util.StringUtils;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+class MessagingPushUtils {
+    private static final String SELF_TAG = "PushNotificationMediaDownloader";
+    private final Context context;
+
+    MessagingPushUtils(final Context context) {
+        this.context = context;
+    }
+
+    Bitmap download(final String url) {
+        Bitmap bitmap = null;
+        try {
+            final URL imageUrl = new URL(url);
+            final HttpURLConnection connection = (HttpURLConnection) imageUrl.openConnection();
+            bitmap = BitmapFactory.decodeStream(connection.getInputStream());
+            connection.disconnect();
```

Comment:
could the networking service be leveraged here rather than writing new networking code? or would it be overkill for this purpose?

Reply:
I am thinking the same too. I dont think there is any negative impact using the HttpURLConnection as well. It might be an overkill.  Moreover, we only have asyn call on network service and that will switch the creating on notification to  network thread.